echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubEventCreator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.pubsub;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hubspot.jinjava.interpret.FatalTemplateErrorsException;
+import com.netflix.spinnaker.echo.artifacts.MessageArtifactTranslator;
+import com.netflix.spinnaker.echo.model.Event;
+import com.netflix.spinnaker.echo.model.Metadata;
+import com.netflix.spinnaker.echo.model.pubsub.MessageDescription;
+import com.netflix.spinnaker.echo.pipelinetriggers.eventhandlers.PubsubEventHandler;
+import com.netflix.spinnaker.echo.pubsub.model.EventCreator;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * An EventCreator that extracts artifacts from an incoming message and creates an event of type pubsub from the
+ * message and artifacts.
+ */
+@Slf4j
+public class PubsubEventCreator implements EventCreator {
+  private final ObjectMapper objectMapper = new ObjectMapper();
+  private final Optional<MessageArtifactTranslator> messageArtifactTranslator;
+
+  public PubsubEventCreator(Optional<MessageArtifactTranslator> messageArtifactTranslator) {
+    this.messageArtifactTranslator = messageArtifactTranslator;
+  }
+
+  public Event createEvent(MessageDescription description) {
+    log.debug("Processing pubsub event with payload {}", description.getMessagePayload());
```

Comment:
While you're refactoring, we should maybe hide this behind a debug log flag to get it hidden even further? This log statement generates a ton of log messages, which are not really helpful after things are working.

Reply:
You mean guard the whole statement by `log.isDebugEnabled()` to avoid evaluating it at all in most cases?  This is already at the debug level, so shouldn't be generating messages unless I'm missing something.

Reply:
I mean have some flag in the pubsub config that turns on this log statement. I think that this log proportionally overwhelms all other debug logs. We tend to run with debug logs enabled, except in echo, because of this statement. It would be nice to have to opt into this very very debug log statement. I don't know if other people experience this pain.

Reply:
Oh, I see---I didn't realize you ran with `debug` enabled, makes sense now.  I'm also happy to just remove the line, do you see any issue in just removing it?

Reply:
I don't! I'd prefer that I think. But I think @jtk54 might have opinions on that

Reply:
This debug message has certainly been useful to us for troubleshooting when we don't have direct access to the spinnaker instance (i.e. most cases). Most users don't have the capability to add logging instrumentation when they need for debugging purposes, so I'm inclined to keep this in. Should we just move it to TRACE?

Reply:
Sounds good, I'll make it `TRACE` instead of just removing it.